src/Grid/grid.jl

```diff
@@ -127,7 +311,65 @@ end
 ##########################
 # Grid utility functions #
 ##########################
-@inline getdim(grid::AbstractGrid{dim}) where {dim} = dim
+"""
+    getneighborhood(top::ExclusiveTopology, grid::Grid{dim,C,T}, cellidx::CellIndex, include_self=false)
+    getneighborhood(top::ExclusiveTopology, grid::Grid{dim,C,T}, faceidx::FaceIndex, include_self=false)
+    getneighborhood(top::ExclusiveTopology, grid::Grid{dim,C,T}, vertexidx::VertexIndex, include_self=false)
+    getneighborhood(top::ExclusiveTopology, grid::Grid{dim,C,T}, EdgeIndex::EdgeIndex, include_self=false)
+
+Returns all directly connected entities of the same type, i.e. calling the function with a `VertexIndex` will return
+a list of directly connected vertices (connected via face/edge). If `include_self` is true, the given `*Index` is included 
+in the returned list.
+
+**Warning:** this feature is highly experimental and very likely subjected to interface changes in the future.
```

Comment:
I think we could highlight this via alert box. (see https://github.com/JuliaDocs/Documenter.jl/blob/cd57c41ab1ff8a8dacf5c21a7d8363b777a6ad9a/src/Documenter.jl#L874-L879)